src/models/dashboard.js

```diff
@@ -9,18 +10,20 @@ export default {
   subscriptions: {
     setup({ dispatch, history }) {
       history.listen(location => {
-        dispatch({
-          type: 'query',
-          payload: location.query,
-        })
+        if (enableQueryData(location.pathname, 'dashboard')) {
+          dispatch({
+            type: 'query',
+            payload: location.pathname.startsWith('/dashboard') ? queryString.parse(location.search) : {},
+          })
+        }
       })
     },
   },
   effects: {
     *query({
       payload,
     }, { call, put }) {
-      const data = yield call(query, parse(payload))
+      const data = yield call(query, queryString.parse(payload))
```

Comment:
Not sure if we still need parse the payload, because it has been parsed before dispatch as above. 

If need to change, please change other places as well.

Reply:
In fact, this parameter will not be wrong even if I don't pass it. Now all the logic of the data filter is done by the front end. Even if I pass the parameter to the backend through the interface, the backend will return the full amount of data. You are right, I should optimize this piece of code for future support of back-end filtering